xcode/iconizer.sh

```diff
@@ -1,52 +1,79 @@
 #!/bin/bash
 set -euo pipefail
 
-# Iconizer shell script by Stefan Herold
+# Iconizer shell script by Stefan Herold and Yunus Tür
 
 # This is a simple tool to generate all necessary app icon sizes from one vector file.
 # To use: specify the path to your vector graphic (PDF format)
-# Example: sh iconizer.sh MyVectorGraphic.pdf
+# Example: sh iconizer.sh VectorGraphic.pdf AppIcon.appiconset
 
 # Requires ImageMagick: http://www.imagemagick.org/
 # Requires jq: https://stedolan.github.io/jq/
 
+# Variables
+
+PDF_PATH=""
+APP_ICON_SET_PATH=""
 CONTENTS_FILE="Contents.json"
+CONTENTS_PATH=""
+IMAGES_LENGTH=0
 
-die () {
-  echo >&2 "$@"
-  exit 1
-}
+# Functions
+
+checkInputs() {
+  ([ $1 != "null" ] && [ $2 != "null" ]) || die "Usage: sh iconizer.sh <IMAGE_PATH.pdf> <APP_ICON_PATH.appiconset>"
+  ([ -e "$1" ] && [ ${1: -4} == ".pdf" ]) || die "Did not find file $1, expected path to an image file with .pdf extension."
+  ([ -e "$2" ] && [ ${2: -11} == ".appiconset" ]) || die "Did not find folder $2, expected path should end with .appiconset extension."
+
+  PDF_PATH=$1
+  APP_ICON_SET_PATH=$2
+  CONTENTS_PATH="$APP_ICON_SET_PATH/$CONTENTS_FILE"
+
+  [ -e "$CONTENTS_PATH" ] || die "Did not find $CONTENTS_PATH. Appiconset must contain a $CONTENTS_FILE file."
 
-[ $# == 2 ] || die "Usage: sh iconizer.sh <file name> <path to *.appiconset>"
-[ -e "$1" ] || die "Did not find file $1, expected path to an image file."
+  IMAGES_LENGTH=$(jq ".images | length" $CONTENTS_PATH)
 
-CONTENT_FILE_PATH="$2/$CONTENTS_FILE"
+  ([ "$IMAGES_LENGTH" != "null" ] && (($IMAGES_LENGTH > 0))) || die "Did not find any images in $CONTENTS_PATH"
+}
+
+generateImages() {
+  echo "Creating icons from $APP_ICON_SET_PATH and updating $CONTENTS_PATH..."
+  for i in $(seq 0 $IMAGES_LENGTH); do
+    generateImage $i
+  done
+  echo "All Done 🎉🎉🎉"
+}
 
-[ -e "$CONTENT_FILE_PATH" ] || die "Did not find $CONTENT_FILE_PATH, expected folder which contains $CONTENTS_FILE"
+generateImage() {
+  image=$(jq ".images[$1]" $CONTENTS_PATH)
+  [ "$image" != "null" ] || break
```

Comment:
Here I get a complaint when I execute this script - it should be return in a function:
```
Creating icons from Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset and updating Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/Contents.json...
Updading Contents.json to appicon_1024.png ✅
Generating appicon_1024.pngconvert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/appicon_1024.png' @ warning/png.c/MagickPNGWarningHandler/1750.
 ✅
---- > /dev/fd/14: line 49: break: only meaningful in a `for', `while', or `until' loop
Updading Contents.json to appicon_0.png ✅
Generating appicon_0.pngconvert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `Targets/DBFFM/Resources/Assets.xcassets/AppIcon.appiconset/appicon_0.png' @ warning/png.c/MagickPNGWarningHandler/1750.
 ✅
All Done 🎉🎉🎉
```

Reply:
This issue is fixed by updating for loop and it is more understandable now.

Reply:
Awesome. Thanks for the contribution 🔥